### ISSUE TITLE
Select count(*) with GROUP BY is very slow 

### ISSUE THREAD

**Issue by let4be** [scylladb/scylla#5807]
Let's say I have a table
```CREATE TABLE links (
    urlFrom text,
    urlTo text,
    domainFrom text,
    domainTo text,

    PRIMARY KEY (urlFrom, urlTo)
);```

I wanna do `SELECT urlFrom, count(urlTo) FROM links GROUP BY urlFrom`
This query is extremely slow and always times out when table becomes big and scylla won't write any warning or anything when executing via cqlsh(like when you ask for filtering without ALLOW FILTERING)

I tried adding `PER PARTITION LIMIT 1 LIMIT 1` with no success...

Scylla version 3.2.1-0.20200122.e3e301906d5
Running on 4.15.0-76-generic(single node, not a cluster)

Is it a bug?

**Comment by dorlaor** [comment]
It times out and slow because this is a complete scan. Even with per
partition limit
you go over all of the partitions. Note that the count should be one with a
limit of one.

You can change the timeout to be long enough in order to wait for the
results.

**Comment by slivne** [comment]
@let4be - since you are grouping scylla needs to scan the rows of partitions, if you are using paging then scylla tries to collect 5000 results / 1M of results and return that.

We maybe should change scylla to return some results up to the timeout to make sure that we less timeout.

For now you can try and reduce the page size - instead of the default 5000 to be 10 and see if this works better for you.

**Comment by dekimir** [comment]
Also, I'm afraid that LIMIT and PARTITION LIMIT currently don't work correctly with GROUP BY. #5361 #5363 

Sorry about that, it's not an easy fix but I'm working on it...

**Comment by haaawk** [comment]
@havaker it seems that your work could help here.

**Comment by avikivity** [comment]
No, GROUP BY means we return more than one row and so cannot parallelize the query.

Probably it's a problem with paging. The coordinator rows coming from the replica, but need to see if it pages to the client in GROUP BY queries.
